File: backend/app/api/airdrop_scanner.py
"""
空投机会扫描器 - API路由
未发币协议扫描 · 交易所活动 · 链上打新 · 测试网 · Twitter大V · 空投资讯 · 机会评分
"""
from fastapi import APIRouter
from app.services.airdrop_scanner_service import AirdropScannerService

router = APIRouter()
svc = AirdropScannerService()


@router.get("/untokenized-protocols")
async def untokenized_protocols():
    """DefiLlama 未发币高TVL协议扫描，带空投概率评分"""
    return svc.get_untokenized_protocols()


@router.get("/exchange-activities")
async def exchange_activities():
    """交易所活动汇总（币安/OKX/Bybit/Gate）"""
    return svc.get_exchange_activities()


@router.get("/launchpad-projects")
async def launchpad_projects():
    """链上打新/IDO项目追踪"""
    return svc.get_launchpad_projects()


@router.get("/testnet-projects")
async def testnet_projects():
    """测试网/激励测试网项目追踪"""
    return svc.get_testnet_projects()


@router.get("/defillama-airdrops")
async def defillama_airdrops():
    """DeFiLlama 已公布空投 + 高概率空投协议"""
    return svc.get_defillama_airdrops()


@router.get("/twitter-kol")
async def twitter_kol_feed():
    """Twitter空投大V推文监控（via RSSHub）"""
    return svc.get_twitter_kol_feed()


@router.get("/news")
async def airdrop_news():
    """空投资讯聚合（12源RSS）"""
    return svc.get_airdrop_news()


@router.get("/opportunity-scores")
async def opportunity_scores():
    """综合机会评分 - 多维度加权排序"""
    return svc.get_opportunity_scores()

File: frontend/src/services/api/airdropScanner.ts
/**
 * 空投机会扫描器 - API客户端
 * 覆盖：DefiLlama + 交易所活动 + 链上打新 + 测试网 + Twitter大V + RSS聚合(12源) + 机会评分
 */
import axios from 'axios'

const BASE = '/api/airdrop-scanner'

export const airdropScannerApi = {
  /** DefiLlama 未发币高TVL协议扫描 */
  getUntokenizedProtocols: () => axios.get(`${BASE}/untokenized-protocols`),

  /** 交易所活动汇总 */
  getExchangeActivities: () => axios.get(`${BASE}/exchange-activities`),

  /** 链上打新/IDO项目追踪 */
  getLaunchpadProjects: () => axios.get(`${BASE}/launchpad-projects`),

  /** 测试网/激励测试网项目追踪 */
  getTestnetProjects: () => axios.get(`${BASE}/testnet-projects`),

  /** DeFiLlama 已公布空投 + 高概率空投协议 */
  getDefiLlamaAirdrops: () => axios.get(`${BASE}/defillama-airdrops`),

  /** Twitter空投大V推文监控 */
  getTwitterKolFeed: () => axios.get(`${BASE}/twitter-kol`),

  /** 空投资讯聚合（12源RSS） */
  getNews: () => axios.get(`${BASE}/news`),

  /** 综合机会评分 */
  getOpportunityScores: () => axios.get(`${BASE}/opportunity-scores`),
}
